allennlp/tests/training/trainer_test.py

```diff
@@ -139,6 +152,59 @@ def test_passing_trainer_multiple_gpus_raises_error(self):
                 self.model, self.optimizer, self.data_loader, num_epochs=2, cuda_device=[0, 1],
             )
 
+    def test_trainer_respects_epoch_size_equals_total(self):
+        num_epochs = 3
+        batches_per_epoch = len(self.data_loader_equal_epoch)
```

Comment:
All of these tests are identical and aren't really testing what you want, because there's nothing actually enforcing that the data loader did what you expected.  You're just reading a property here and are using this value in an assert below, in a way that should always be true of the trainer and is already tested.  I'd expect something like `assert batches_per_epoch == 3` here (or whatever the value should be).  Same for the tests below.